### PR TITLE
go: Improve integration test failure

### PR DESCRIPTION
**Pull request by lann**
The integration test times out after 10 minutes if spin fails to start. This checks for spin exiting before the test is over.

### CODE REVIEW

File: sdk/go/integration_test.go [hunk 1]
```diff
@@ -17,14 +17,18 @@ const spinBinary = "../../target/debug/spin"
 func retryGet(t *testing.T, url string) *http.Response {
 	t.Helper()
 
-	const maxTries = 120 // (10min/5sec)
+	const maxTries = 600 // (10min)
```
Comment (rylev) [inline sdk/go/integration_test.go:20]: Why do we try for so long?

Comment (lann) [reply sdk/go/integration_test.go:20]: :shrug:

https://github.com/fermyon/spin/blob/3f7c2fee0c22fe24ea4fc74fd516142b02936cce/sdk/go/integration_test.go#L44-L45

Comment (adamreese) [reply sdk/go/integration_test.go:20]: The timeout is long because we need to wait for the build also.  TinyGo can take a bit of time in CI

**Review by adamreese — APPROVED** [review]
We might need to increase timeout again but LGTM

**Pull request merged**